cmp/StoreFilterField.js

```diff
@@ -51,7 +57,7 @@ export class StoreFilterField extends Component {
         let searchTerm = escapeRegExp(this.value);
 
         let filter = null;
-        if (searchTerm && fields.length) {
+        if (searchTerm && fields && fields.length) {
```

Comment:
We should note fields as required above and avoid this check.  Configuring it without fields seems like an error to me.